Assets/Scripts/Environment/Hangar/BatterySpawnPositionManager.cs

```diff
@@ -0,0 +1,41 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class BatterySpawnPositionManager : MonoBehaviour
+{
+    [SerializeField] private GameObject[] spawnPositions;
+
+    [SerializeField] private GameObject[] batteries;
+
+    public void Awake()
+    {
+        // As the game starts, spawn the batteries at their initial position
+        for (int i = 0; i < batteries.Length; i++)
+        {
+            batteries[i].transform.position = spawnPositions[i].transform.position;
+        }
+    }
+
+    // Respawn the battery when the player tries to take it out of the hangar
+    public void RespawnBattery(Transform objectToMove, Collider boundaries)
+    {
+        Collider[] colliders;
+
+        // We loop through the spawn positions until we find one that is free
+        for (int i = 0; i < spawnPositions.Length; i++)
+        {
+            // Check if there is already a gameobject within the space where we want to spawn the battery
+            colliders = Physics.OverlapBox(spawnPositions[i].transform.position, boundaries.bounds.extents, Quaternion.identity);
+
+            // If Physics.OverlapBox() returns nothing, it means that the space is free
+            if(colliders.Length == 0)
+            {
+                objectToMove.transform.position = spawnPositions[i].transform.position;
+                return;
+            }
+        }
+
+        Debug.LogError("There are no available battery spawn positions. We need to add more.");
```

Comment:
What happens here if there are no slots available? The player will keep holding the battery right? Maybe we should make the battery drop next to door or something, I'm not sure. It's not a huge deal at this point though.

Reply:
So the battery does get reparented but it can be dropped in the office at the moment if there are no slots free. I think that's fine though as we'll make sure we have enough slots. Which was the whole point of this. 

I like that you log the error.

Reply:
What you mentioned is why logged the error. I want the game to stop when those bugs arise so that you can take notice.